core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/BasicLoadBalancingPolicy.java

```diff
@@ -321,6 +332,31 @@ protected Set<Node> getReplicas(@Nullable Request request, @Nullable Session ses
       return Collections.emptySet();
     }
 
+    if (table != null && tabletMap != null) {
+      Partitioner p =
+          (partitioner != null
+              ? partitioner
+              : context
+                  .getTokenFactoryRegistry()
+                  .tokenFactoryFor(maybeTokenMap.get().getPartitionerName()));
+      Token t = (token != null ? token : p.hash(key));
+      if (t instanceof TokenLong64) {
+        Map<UUID, Node> currentNodes = session.getMetadata().getNodes();
+        Set<UUID> replicaUUIDs =
+            tabletMap.getReplicas(
+                keyspace.asInternal(), table.asInternal(), ((TokenLong64) t).getValue());
+        Set<Node> replicas = new HashSet<>();
+        for (UUID replica : replicaUUIDs) {
+          if (currentNodes.containsKey(replica)) {
+            replicas.add(currentNodes.get(replica));
+          }
+        }
+        if (!replicas.isEmpty()) {
+          return replicas;
+        }
+      }
```

Comment:
I think we should distinguish 2 cases here:
- Table is a Tablet table (so it is present in `tabletMap`) , but we don't have information for this Token. In this case imo we shouldn't fall back to `TokenMap` routing because it is totally irrelevant to a tablet table
- Table is vnode-based table: in this case yes, we should fall back to  `TokenMap` routing

Reply:
Makes sense, but I've left this flow as it is for now. I'd like to fix that with later expanding `KeyspaceMetadata` and `TableMetadata` classes, but that is probably something more suitable for another PR.
Right now stupid fallbacks should happen only once for each tablet.